### PR TITLE
added spendLimit mediation

### PR DESCRIPTION
**Pull request by azhamn**

### CODE REVIEW

File: com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/mediator/CheckLimitMediator.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        try {
+            GroupEventUnmarshaller.startGroupEventUnmarshaller(xml);
+        } catch (JAXBException e) {
+            return null;
+        }
+
+        return GroupEventUnmarshaller.getInstance();
+    }
+
+    public boolean mediate(MessageContext mc) throws CustomException {
```
Comment (rajithsiriwardana) [inline com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/mediator/CheckLimitMediator.java]: Throwing the custom exception. Where are you handling those thrown exceptions

File: com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/mediator/UserInfoMediator.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+public class UserInfoMediator extends AbstractMediator {
+    public boolean mediate(MessageContext messageContext) {
+
+        try {
+            String operatorName = messageContext.getProperty("OPERATOR_NAME").toString();
+            String consumerKey = messageContext.getProperty("CONSUMER_KEY").toString();
+            String spendLimitConfig = messageContext.getProperty("spendLimitConfig").toString();
+            messageContext.setProperty("userInfoEnabled", getGroupUserInfo(operatorName, consumerKey,spendLimitConfig));
+        } catch (Exception e) {
+        //do error handling here
```
Comment (rajithsiriwardana) [inline com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/mediator/UserInfoMediator.java]: Todo items

**Comment by azhamn** [comment]
@rajithsiriwardana This is a WIP. I have only done basic implementations so there is less of a overhead for the person taking over. Please sync with @shazni.

File: com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/mediator/CheckLimitMediator.java (hunk 1 above)
Comment (shazni) [inline com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/mediator/CheckLimitMediator.java]: The entire CustomException class is not needed. In the new implementation when errorneous conditions happen it has to be handled separately.

File: com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/CustomException.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package com.wso2telco.dep.common.mediation.spendlimit;
+
+
+public class CustomException extends RuntimeException {
```
Comment (shazni) [inline com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/CustomException.java]: This entire class needs to be removed. Exceptions are handled differently in synapse

File: com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/mediator/CheckLimitMediator.java (hunk 1 above)
Comment (shazni) [inline com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/spendlimit/mediator/CheckLimitMediator.java]: These custom exceptions have to be removed.

**Pull request closed without merging**